src/sinks/util/buffer/metrics/split.rs

```diff
@@ -0,0 +1,320 @@
+use std::collections::VecDeque;
+
+use vector_core::event::{metric::MetricData, Metric, MetricValue};
+
+#[allow(clippy::large_enum_variant)]
+enum SplitState {
+    Single(Option<Metric>),
+    Multiple(VecDeque<Metric>),
+}
+
+/// An iterator that returns the result of a metric split operation.
+pub struct SplitIterator {
+    state: SplitState,
+}
```

Comment:
That's a fair observation, and yeah, the `Iterator` implementation _could_ live on `SplitState` itself. I mostly wanted to encapsulate the state to avoid outside manipulation.

I'm not sure if there's a way to put the iterator on the enum while making it impossible for callers to futz with the internal state outside of the `Iterator` interface?

Reply:
Oh, hmm, yeah, because enum members are always pub. Fair enough.